code/game/objects/structures/beds_chairs/chair.dm

```diff
@@ -437,3 +437,42 @@
 
 /obj/structure/chair/mime/post_unbuckle_mob(mob/living/M)
 	M.pixel_y -= 5
+
+
+/obj/structure/chair/plastic
+	icon_state = "plastic_chair"
+	name = "folding plastic chair"
+	desc = "No matter how much you squirm, it'll still be uncomfortable."
+	resistance_flags = FLAMMABLE
+	max_integrity = 50
+	custom_materials = list(/datum/material/plastic = 2000)
+	buildstacktype = /obj/item/stack/sheet/plastic
+	buildstackamount = 2
+	item_chair = /obj/item/chair/plastic
+
+/obj/structure/chair/plastic/post_buckle_mob(mob/living/Mob)
+	Mob.pixel_y += 2
+	.=..()
+	if (Mob.nutrition >= NUTRITION_LEVEL_FAT)
```

Comment:
Buckle could be called off various ss calls so you probably want to wrap this in a timer or invokeasync.

Reply:
Right, I believe I got this working. No runtimes and the behavior's looking right.